### ISSUE TITLE
Error enum values not propagated as exit codes for benchmark comparison in CI.

### ISSUE THREAD

**Issue by loonatick-src** [ordo-one/package-benchmark#235]
From the docs:
> These baselines can then be checked with:
> 
> ```
> swift package benchmark baseline check --check-absolute-path /relative/or/absolute/path/to/Thresholds
> ```
> 
> The absolute check will have an exit code of 0 if the check is exactly equal, but will return with exit code 2 if there were any regressions or exit code 4 if there were only improvements.

But, this does not seem to be the case; this is what I saw.

```
$ swift package benchmark baseline check main merge_request              
Building for debugging...
[1/1] Write swift-version-6E0725D62189FA0A.txt
Build complete! (0.29s)
Building for debugging...
[1/1] Write swift-version-6E0725D62189FA0A.txt
Build complete! (0.66s)
Build complete!
Building BenchmarkTool in release mode...

=======================================================================================
Threshold deviations for test benchmarks:TestBenchmarks
=======================================================================================
╒══════════════════════════════════════════╤═════════════════╤═════════════════╤═════════════════╤═════════════════╕
│ Time (wall clock) (μs, %)                │            main │   merge_request │    Difference % │     Threshold % │
╞══════════════════════════════════════════╪═════════════════╪═════════════════╪═════════════════╪═════════════════╡
│ p75                                      │             185 │             170 │              -8 │               5 │
╘══════════════════════════════════════════╧═════════════════╧═════════════════╧═════════════════╧═════════════════╛

New baseline 'merge_request' is BETTER than the 'main' baseline thresholds.

error: benchmarkThresholdImprovement
$ echo $?
1
```

The return code for improvement is 1, and from the docs I expected it to be 4 in the CI.

I see that the error enum's value is defined as 4 [here](https://github.com/ordo-one/package-benchmark/blob/e495d16b054737222106f51c80534a020490ac3d/Plugins/BenchmarkCommandPlugin/BenchmarkCommandPlugin.swift#L426), and it is thrown from [`performCommand`](https://github.com/ordo-one/package-benchmark/blob/e495d16b054737222106f51c80534a020490ac3d/Plugins/BenchmarkCommandPlugin/BenchmarkCommandPlugin.swift#L396).

I suspect that `performCommand`'s caller might not be calling `exit` with the error enum's raw value as done [here](https://github.com/apple/swift-package-manager/blob/8268caa411b45fde7d886c2e7fc7982ae277040c/Sources/PackagePlugin/Plugin.swift#L134).

```swift
 Diagnostics.error(String(describing: error))
 exit(1)
```

This seems to be happening in the example benchmark comparison run. I couldn't completely trace the call across library boundaries. Is this a bug or am I missing something?

# Environment
- OS: macOS Sonoma 14.4
- package-benchmark version: 1.22.3
- Swift version: 5.11-dev, DEVELOPMENT-SNAPSHOT-2023-12-07-a

**Comment by hassila** [comment]
Yeah, it's the same issue. It's possible to work around by running the benchmark tool directly without swiftpm - running the benchmark with --debug you can see the actual command line required to drive the benchmark tool and instead run that in ci, the you should get the correct error code surfaced.

**Comment by loonatick-src** [comment]
Is this also reflected in the docs? I could not find anything of the sort and was going entirely by the 0/2/4 exit codes.

**Comment by hassila** [comment]
Thanks for the report, unfortunately away for a couple of weeks and won't have time to check on this until then, sorry for delay.

**Comment by loonatick-src** [comment]
No worries! Also thank you for pointing out the workaround
